### PR TITLE
Updating stubs in java.nio

### PR DESCRIPTION
**Pull request by albertomisail**
Due to missing sources and dependencies some of the methods in some classes had to be commented out. Now that FileSystemProvider can be added, some of the methods can be readded

### CODE REVIEW

File: jre_emul/jre_sources.mk [hunk 1]
```diff
@@ -1095,6 +1101,7 @@ JAVA_PUBLIC_SOURCES_FILE = \
   java/nio/file/StandardCopyOption.java \
   java/nio/file/StandardOpenOption.java \
   java/nio/file/StandardWatchEventKinds.java \
+  java/nio/file/TempFileHelper.java \
```
Comment (tomball) [inline jre_emul/jre_sources.mk:1104]: Should be in JAVA_PRIVATE_SOURCES_FILE.

File: jre_emul/jre_sources.mk [hunk 2]
```diff
@@ -1064,6 +1064,7 @@ JAVA_PUBLIC_SOURCES_FILE = \
   java/nio/file/ClosedDirectoryStreamException.java \
   java/nio/file/ClosedFileSystemException.java \
   java/nio/file/ClosedWatchServiceException.java \
+  java/nio/file/CopyMoveHelper.java \
```
Comment (tomball) [inline jre_emul/jre_sources.mk:1067]: Should be in JAVA_PRIVATE_SOURCES_FILE.

File: jre_emul/jre_sources.mk [hunk 3]
```diff
@@ -1075,9 +1076,13 @@ JAVA_PUBLIC_SOURCES_FILE = \
   java/nio/file/FileSystemException.java \
   java/nio/file/FileSystemLoopException.java \
   java/nio/file/FileSystemNotFoundException.java \
+  java/nio/file/FileSystems.java \
+  java/nio/file/FileTreeIterator.java \
```
Comment (tomball) [inline jre_emul/jre_sources.mk:1080]: Should be in JAVA_PRIVATE_SOURCES_FILE.

File: jre_emul/jre_sources.mk [hunk 4]
```diff
@@ -1075,9 +1076,13 @@ JAVA_PUBLIC_SOURCES_FILE = \
...
+  java/nio/file/FileTreeWalker.java \
```
Comment (tomball) [inline jre_emul/jre_sources.mk:1081]: Should be in JAVA_PRIVATE_SOURCES_FILE.

File: jre_emul/jre_sources.mk (hunk 3 above)
Comment (albertomisail) [reply jre_emul/jre_sources.mk:1080]: Does this refer to both sources or only FileTreeIterator?

Comment (tomball) [reply jre_emul/jre_sources.mk:1080]: Just FileTreeIterator, as I made a separate comment for each package-private class. Basically, all package-private classes go in PRIVATE, all public and protected in PUBLIC.

**Pull request merged**